jme3-core/src/main/java/com/jme3/anim/tween/action/BaseAction.java

```diff
@@ -6,10 +6,35 @@
 
 import java.util.List;
 
+/**
+ * A Concrete fast Implementation for the interface #{@link Action},
+ * used to create a subset of actions gathered from the extracted actions of the assigned tween.
+ *
+ * Example of operation :
+ * <pre class="prettyprint">
+ * //create a base action from a tween.
+ * final BaseAction action = new BaseAction(Tweens.parallel(clipAction0, clipAction1));
+ * //set the action properties - utilized within the #{@link Action} class.
+ * baseAction.setLength(10f);
+ * baseAction.setSpeed(2f);
+ * //register the action as an observer to the animComposer control.
+ * animComposer.addAction("basicAction", action);
+ * //make a new Layer for a basic armature mask
+ * animComposer.makeLayer(ActionState.class.getSimpleName(), new ArmatureMask());
+ * //run the action within this layer
+ * animComposer.setCurrentAction("basicAction", ActionState.class.getSimpleName());
+ * </pre>
+ * <b>Created by Nehon.</b>
+ */
 public class BaseAction extends Action {
 
     final private Tween tween;
 
+    /**
+     * Gathers the actions from a tween into a subset collection #{@link Action#actions},
+     * which joins anim actions hierarchy later on when instantiated by #{@link BlendableAction} implementation entites.
```

Comment:
What are "implementation entites"?
The abstract language used here obscures the key information, namely "How is this software useful?"

Reply:
That's better.